FakeHttpService.Tests/Given_I_want_to_get_fake_responses_in_a_test.cs

```diff
@@ -15,28 +17,159 @@ namespace FakeHttpService.Tests
+                .WhereUri(uri => uri.ToString().EndsWith("customapicall"))
+                .Then()
+                .RespondWith(async r => {
+                    r.StatusCode = 200;
+                    await Task.Yield();
+                }))
+            {
+                var response = await new HttpClient().GetAsync( new Uri(fakeService.BaseAddress, "/customapicall"));
+
+                response.EnsureSuccessStatusCode();
+            }
+        }
+
+        [Fact]
+        public async Task When_filtering_method_Then_the_expected_response_is_returned()
+        {
+            using (var fakeService = new FakeHttpService()
+                .OnRequest()
+                .WhereMehtod(method => method == "POST")
+                .Then()
+                .RespondWith(async r =>
+                {
+                    r.StatusCode = 200;
+                    await Task.Yield();
+                })
+            .OnRequest(_ => true).RespondWith(async r =>
+                {
+                    r.StatusCode = 500;
+                    await Task.Yield();
+                }))
+            {
+                var response = await new HttpClient().PostAsync(new Uri(fakeService.BaseAddress, "/customapicall"), new StringContent("nothing"));
+
+                response.EnsureSuccessStatusCode();
+
+                 response = await new HttpClient().GetAsync(new Uri(fakeService.BaseAddress, "/customapicall"));
+
+                response.StatusCode.Should().Be(HttpStatusCode.InternalServerError);
+            }
+        }
+
+        [Fact]
+        public async Task When_filtering_body_Then_the_expected_response_is_returned()
+        {
+            using (var fakeService = new FakeHttpService()
+                .OnRequest()
+                .WhereBodyAsString(body => body == "nothing")
+                .Then()
+                .RespondWith(async r =>
+                {
+                    r.StatusCode = 200;
+                    await Task.Yield();
+                })
+                .OnRequest(_ => true).RespondWith(async r =>
+                {
+                    r.StatusCode = 500;
+                    await Task.Yield();
+                }))
+            {
+                var response = await new HttpClient().PostAsync(new Uri(fakeService.BaseAddress, "/customapicall"), new StringContent("nothing"));
+
+                response.EnsureSuccessStatusCode();
+
+                response = await new HttpClient().PostAsync(new Uri(fakeService.BaseAddress, "/customapicall"), new StringContent("different"));
+
+                response.StatusCode.Should().Be(HttpStatusCode.InternalServerError);
+            }
+        }
+
+        [Fact]
+        public async Task When_filtering_body_as_json_Then_the_expected_response_is_returned()
+        {
+            using (var fakeService = new FakeHttpService()
+                .OnRequest()
+                .WhereBodyAsJson(body => JToken.DeepEquals( body, JToken.Parse("{ field: 1}")))
```

Comment:
In addition, or even in stead of, I'd provide WhereBodyFromJson<T>. Wouldn't you rather work with a deserialized POCO?